crates/full-node/sov-sequencer/src/preferred/nonce_buffer_task.rs

```diff
@@ -33,12 +37,64 @@ use super::Confirmation;
 // large TXs. Perhaps we should consider enforcing batch size limits before the nonce check in accept_tx()?
 const MAX_BUFFERED_TXS: usize = 20_000;
 
+// Maximum depth of the main input queue for the nonce buffer task.
+// This is separate from MAX_BUFFERED_TXS because the main input queue handles more than just
+// buffered transactions (e.g., TxExecuted, TxPersisted messages). Currently set to the same
+// value pending real-world testing.
+const MAX_BUFFER_INPUT_QUEUE: usize = 20_000;
+
+/// Send a message to an mpsc channel with metrics tracking.
+///
+/// Uses try_send first for the non-blocking fast path. If the channel is full, falls back to
+/// blocking send and records how long the send was blocked.
+///
+/// Returns Ok(()) on success, or Err if the channel is closed (receiver dropped).
+async fn send_with_metrics<T, M: ChannelQueueMetricName + 'static>(
+    sender: &mpsc::Sender<T>,
+    message: T,
+) -> Result<(), mpsc::error::SendError<T>> {
+    let mut metrics = ChannelQueueMetrics::<M>::default();
+
+    let result = match sender.try_send(message) {
+        Ok(()) => Ok(()),
+        Err(mpsc::error::TrySendError::Full(message)) => {
+            tracing::trace!(
+                "{} is full. Blocking until capacity available.",
+                M::MEASUREMENT_NAME
+            );
+            let started_blocking = Instant::now();
+            let result = sender.send(message).await;
+            metrics.blocked_for_us = started_blocking.elapsed().as_micros() as u64;
+            result
+        }
+        Err(mpsc::error::TrySendError::Closed(message)) => Err(mpsc::error::SendError(message)),
+    };
+
+    if result.is_ok() {
+        metrics.queue_depth = sender.max_capacity() - sender.capacity();
+        sov_metrics::track_metrics(|t| {
+            t.submit(metrics);
+        });
+    }
```

Comment:
This is going to result in a *lot* of metrics - one for each transaction. Maybe we can aggregate the way we do with the sequencer inner? What we do there is pre-allocate batch of 128 metrics in a Vec, and then impl `Metric` for that vec so that we only have to submit once per 129 metrics

Reply:
Good point - I'll go ahead and implement that before merging